### PR TITLE
Initialize health check gateway state to Unknown

### LINKED ISSUE #6747: Possible behavior change in the /services/SERVICE_NAME/SERVICE_GROUP/health API
On Hab 0.82.0 I've observed a service whose health check was stuck in the unknown state.  Now, while I don't know why that is the case, I also observed a small behavior change in the HTTP response code from the `/services/SERVICE_NAME/SERVICE_GROUP/health` API endpoint.

Here is the state of the process:

```
[231][default:/src:0]# curl -v http://localhost:9631/services/deployment-service/default 2>/dev/null | jq .health_check
"Unknown"
[232][default:/src:0]# curl -v http://localhost:9631/services/deployment-service/default 2>/dev/null | jq .process
{
  "pid": 28029,
  "state": "up",
  "state_entered": 1563456058
}
```

and the health check returns 404:

```
# curl -v http://localhost:9631/services/deployment-service/default/health
*   Trying 127.0.0.1:9631...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 9631 (#0)
> GET /services/deployment-service/default/health HTTP/1.1
> Host: localhost:9631
> User-Agent: curl/7.65.1
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 404 Not Found
< content-length: 0
< date: Fri, 19 Jul 2019 13:49:02 GMT
< 
* Connection #0 to host localhost left intact
```

I believe this is likely a behavior change because we had previously observed that HTTP 404 was reserved for unloaded services while HTTP 500 was returned for loaded services with an unknown health-check status.  You can see the mapping of API responses to their semantic meanings in this client code that used to work:

https://github.com/chef/automate/blob/master/components/automate-deployment/pkg/habapi/habapi.go#L132-L143

which is also in line with the documentation here:

https://github.com/habitat-sh/habitat/blob/feb04cdd07500207b9c4ffad03acbccedb3f7910/components/sup/doc/api.raml#L248-L262

If this behavior change was intended we should probably update the documentation.  However, until #4988 is fixed, the /heatlh API is the only way to get accurate health information and this change makes it hard to distinguish an unloaded service from an unknown service.

- [x] The OS (including version) where you are running any of the Habitat commands: Linux 4.9.125 with enough hab packages to vaguely look like an operating system.

- [x] Debug/backtrace of the command you are trying to run. You can set the following environment variables before running the `hab` command to generate a trace: I don't think this will be helpful.

- [x] Current Habitat environment variables where the `hab` command or supervisor is running.  These can be gathered using:

```
# ps | grep hab-sup
 4413 pts/0    00:00:38 hab-sup
# tr '\0' '\n' < /proc/4413/environ 
CGO_ENABLED=0
CHEF_AUTOMATE_SKIP_SYSTEMD=true
CHEF_CHANNEL=dev
CHEF_DEV_ENVIRONMENT=true
DEVPROXY_URL=host.docker.internal
DS_DEV=true
ELASTICSEARCH_URL=http://localhost:10141
ES_VERSION=5.6.4
EXPEDITOR_TARGET_CHANNEL=current
GATEWAY_HOST=localhost
GATEWAY_PORT=2000
GATEWAY_URL=https://localhost:2000
GITHUB_TOKEN=cantfoolmehabteamyoujustwantedmypassword
GOBIN=/hab/cache/src/scaffolding-go-gopath/bin
GOPATH=/hab/cache/src/scaffolding-go-gopath
GO_DEBUG_PORT=2345
GO_DEP_VERSION=0.5.0
HAB_HART_COMPRESSION_LEVEL=0
HAB_LAUNCHER_PID=4349
HAB_LAUNCHER_PIPE=/tmp/rust-ipc-socket.jFPpSk
HAB_LICENSE=accept-no-persist
HAB_ORIGIN=ssd
HAB_STUDIO_SECRET_GITHUB_TOKEN=cantfoolmehabteamyoujustwantedmypassword
HAB_STUDIO_SUP=false
HOME=/root
HOSTNAME=a3d05aff8d6d
INGEST_HOST=localhost
INGEST_PORT=10122
INGEST_URL=https://localhost:10122
LS_COLORS=rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:
OLDPWD=/hab/cache/src/scaffolding-go-gopath/src/github.com/chef/automate/components/automate-cli
PATH=/hab/pkgs/core/hab-plan-build/0.82.0/20190605215512/bin:/hab/pkgs/core/diffutils/3.6/20190115013221/bin:/hab/pkgs/core/less/530/20190115013008/bin:/hab/pkgs/core/make/4.2.1/20190115013626/bin:/hab/pkgs/core/mg/20180408/20190115015655/bin:/hab/pkgs/core/util-linux/2.32/20190115013746/bin:/hab/pkgs/core/vim/8.1.0577/20190115015449/bin:/hab/pkgs/core/ncurses/6.1/20190115012027/bin:/hab/pkgs/core/acl/2.2.53/20190115012136/bin:/hab/pkgs/core/attr/2.4.48/20190115012129/bin:/hab/pkgs/core/bash/4.4.19/20190115012619/bin:/hab/pkgs/core/binutils/2.31.1/20190115003743/bin:/hab/pkgs/core/bzip2/1.0.6/20190115011950/bin:/hab/pkgs/core/coreutils/8.30/20190115012313/bin:/hab/pkgs/core/file/5.34/20190115003731/bin:/hab/pkgs/core/findutils/4.6.0/20190115013303/bin:/hab/pkgs/core/gawk/4.2.1/20190115012752/bin:/hab/pkgs/core/glibc/2.27/20190115002733/bin:/hab/pkgs/core/grep/3.1/20190115012541/bin:/hab/pkgs/core/gzip/1.9/20190115013612/bin:/hab/pkgs/core/hab/0.82.0/20190605214032/bin:/hab/pkgs/core/libcap/2.25/20190115012150/bin:/hab/pkgs/core/openssl-fips/2.0.16/20190115014207/bin:/hab/pkgs/core/openssl/1.0.2r/20190305210149/bin:/hab/pkgs/core/pcre/8.42/20190115012526/bin:/hab/pkgs/core/rq/0.10.4/20190115014520/bin:/hab/pkgs/core/sed/4.5/20190115012152/bin:/hab/pkgs/core/tar/1.30/20190115012709/bin:/hab/pkgs/core/unzip/6.0/20190115014516/bin:/hab/pkgs/core/wget/1.19.5/20190305211748/bin:/hab/pkgs/core/xz/5.2.4/20190115013348/bin:/bin:/hab/cache/src/scaffolding-go-gopath/bin
PLAN_CONTEXT=/src/habitat
POSTGRESQL_URL=localhost:5432
PWD=/src
SHLVL=2
STUDIO_TYPE=default
SUPPORTED_HAB_VERSION=0.77.0
TERM=xterm
TERMINFO=/hab/pkgs/core/ncurses/6.1/20190115012027/share/terminfo
UI_LOG_FILE=/tmp/automate-ui.log
_=/hab/pkgs/core/hab/0.82.0/20190605214032/bin/hab
scaffolding_go_base_path=github.com/chef
scaffolding_go_gopath=/hab/cache/src/scaffolding-go-gopath
scaffolding_go_pkg_path=/hab/cache/src/scaffolding-go-gopath/src/github.com/chef/automate
scaffolding_go_workspace_src=/hab/cache/src/scaffolding-go-gopath/src
```

### PR DESCRIPTION
**Pull request by stevendanna**
Previously, the gateway state had no health check information for a
given service until the first health-check. This resulted in a 404 for
the `/services/SERVICE_NAME/SERVICE_GROUP/health` endpoint. However,
according to the documentation, a 404 response code means the service
isn't loaded:

```
 /{name}/{group}/health:
     get:
         description: Health check status and output for the given service group
         responses:
             200:
                 description: Health Check - Ok / Warning
                 body:
                     application/json:
                         type: healthCheckOutput
             404:
                 description: Service not loaded
             500:
                 description: Health Check - Unknown
             503:
                 description: Health Check - Critical
```

This change avoids the problem by populating the gateway state with
Unknown when we start the health checks. As a result we now return 500
for the health endpoint of a loaded service until the first
health-check comes back, in accordance with the API documentation.

Fixes #6747 

REVIEWER NOTES:

- I'm not actually sure this is the best solution. It is a bit hard to follow all of the conditions in which we might call start_health_checks.  We could instead check for the existence of the health check directly in http_gateway.rs by checking `service_from_services(&service_group, state.gateway_state.lock_gsr().services_data())`

### CODE REVIEW

File: components/sup/src/manager/service.rs [hunk 1]
```diff
@@ -53,8 +53,11 @@ pub use habitat_common::templating::{config::{Cfg,
                                                PkgProxy}};
 use habitat_common::{outputln,
                      templating::{config::CfgRenderer,
-                                  hooks::Hook,
-                                  package::DEFAULT_USER}};
+                                  hooks::Hook}};
+
+#[cfg(windows)]
+use habitat_common::templating::package::DEFAULT_USER;
```
Comment (stevendanna) [inline components/sup/src/manager/service.rs]: This and some of the unrelated looking changes are to get rid of warnings when compiling on linux.

**Comment by christophermaier** [comment]
@stevendanna Which issue does this address? #674 doesn't seem like the right one.

**Comment by christophermaier** [comment]
@stevendanna You might want to rebase this once #7088 lands, since it takes care of the compiler warnings you found.

**Comment by stevendanna** [comment]
Sorry, correct fix issue is https://github.com/habitat-sh/habitat/issues/6747

I'll fix that up in the commit message when I rebase on top of that change you mentioned.
